### PR TITLE
Fix - Dataflows for Java Lambda

### PR DESCRIPTION
**Pull request by khemrajrathore**
The PR fixes the following 
- Flows which originate from inside of lambda and sink is outside of lambda
- The above is achieved by linking the MethodRef of lambda with the corresponding MethodReturn node in dataflow engine

### CODE REVIEW

File: dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala [hunk 1]
```diff
@@ -93,49 +101,71 @@ class TaskCreator(context: EngineContext) {
...
-            ReachableByTask(taskStack, newPath)
-          }
-        } else {
-          returnStatements.map { returnStatement =>
-            val newPath = Vector(PathElement(methodReturn, result.callSiteStack)) ++ path
-            val taskStack =
-              result.taskStack :+ TaskFingerprint(returnStatement, call :: result.callSiteStack, callDepth + 1)
-            ReachableByTask(taskStack, newPath)
+      outArg.toList.flatMap {
+        case call: Call =>
```
Comment (fabsx00) [inline dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala:105]: This first part looks to me as if the `.collect` was just turned into an equivalent `.flatMap`. Can you explain if that is the only change up to line 130?

File: dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala [hunk 2]
```diff
@@ -93,49 +101,71 @@ class TaskCreator(context: EngineContext) {
...
-          List[MethodParameterOut]()
-        } else {
-          argToOutputParams(arg).l
-        }
-        outParams
-          .filterNot(_.method.isExternal)
-          .map { p =>
-            val newStack = arg.inCall.headOption.map { x => x :: result.callSiteStack }.getOrElse(result.callSiteStack)
-            ReachableByTask(result.taskStack :+ TaskFingerprint(p, newStack, callDepth + 1), path)
+      args.toList.flatMap {
```
Comment (fabsx00) [inline dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala:133]: From here to line 150, are there any changes or is this just formatting?

**Review by fabsx00 — COMMENTED** [review]
Looks good: just seems like there are a lot of mere format changes. Can you revise this slightly so that the semantics of this change become more apparent?

File: dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala (hunk 1 above)
Comment (khemrajrathore) [reply dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala:105]: Yes, only formatting and handling of `case _ => Vector.empty` for nonCall nodes

File: dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala (hunk 2 above)
Comment (khemrajrathore) [reply dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala:133]: Yes, only formatting and handling of `case _ => Vector.empty` for nonExpression nodes

**Review by fabsx00 — APPROVED** [review]

**Pull request merged**